benchmarks/allocation/alloc.cc

```diff
@@ -9,12 +9,13 @@ using Debug = ConditionalDebug<DEBUG_ALLOCBENCH, "Allocator benchmark">;
  * Try allocating 1 MiB of memory in allocation sizes ranging from 32 - 4096
  * bytes, report how long it takes.
  */
-void __cheri_compartment("allocbench") run()
+int __cheri_compartment("allocbench") run()
 {
 	// Make sure sail doesn't print annoying log messages in the middle of the
 	// output the first time that allocation happens.
 	free(malloc(16));
-	heap_quarantine_empty();
+	Debug::Invariant(heap_quarantine_empty() != -ECOMPARTMENTFAIL,
```

Comment:
```suggestion
	Debug::Invariant(heap_quarantine_empty() == 0,
```
This invariant is not strong enough as, according to the comment on `heap_quarantine_empty`, other failures such as `ENOTENOUGHSTACK` and `ENOTENOUGHTRUSTEDSTACK` are possible. This applies to all other calls to this and perhaps other compartment calls?

Reply:
I think we'd like to know about any failures so would prefer the stronger invariant.